notifications_utils/recipients.py

```diff
@@ -289,15 +289,20 @@ def is_optional_address_column(self, key):
         )
 
     @property
-    def has_recipient_columns(self):
+    def required_recipient_column_headers(self):
         return set(
             Columns.make_key(recipient_column)
             for recipient_column in self.recipient_column_headers
             if not self.is_optional_address_column(recipient_column)
-        ) <= set(
-            self.column_headers_as_column_keys
         )
 
+    @property
+    def has_recipient_columns(self):
+        # self.column_headers_as_column_keys is an OrderedSet. We need
+        # to cast it to a set because doing subset operations on
+        # OrderedSet returns `True` when you wouldn’t expected it to
```

Comment:
Optional: Maybe change to : `OrderedSet` has misleading behaviour when comparing it to `Set` objects

Reply:
Done, and added a link to the relevant issue (squashed into 30741d9)